pkg/cloud/services/securitygroup/securitygroups.go

```diff
@@ -162,6 +162,12 @@ func (s *Service) ReconcileSecurityGroups() error {
 
 func (s *Service) DeleteSecurityGroups() error {
 	conditions.MarkFalse(s.scope.InfraCluster(), infrav1.ClusterSecurityGroupsReadyCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
+
+	if s.scope.VPC().ID == "" {
+		conditions.MarkFalse(s.scope.InfraCluster(), infrav1.ClusterSecurityGroupsReadyCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
Cluster is not yet being deleted here:
```suggestion
		conditions.MarkFalse(s.scope.InfraCluster(), infrav1.ClusterSecurityGroupsReadyCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
```

Reply:
Actually we are not setting cluster deletion here. We are just setting securityGroup condition to be deleted as the security group does not exist.

Reply:
You are right.